wire-ios-data-model/Tests/MLS/MLSServiceTests.swift

```diff
@@ -1086,6 +1086,70 @@ class MLSServiceTests: ZMConversationTestsBase, MLSServiceDelegate {
         XCTAssertEqual(groupInfoInvocations.count, 0)
     }
 
+    // MARK: - Handling out of sync conversations
+
+    func test_RepairOutOfSyncConversations_RejoinsOutOfSyncConversation() {
+        // GIVEN
+        let conversationAndOutOfSyncTuples = [
+            createConversation(outOfSync: true),
+            createConversation(outOfSync: true),
+            createConversation(outOfSync: false)
+        ]
+
+        // mock conversation epoch
+        mockCoreCrypto.mockConversationEpoch = { groupID in
+            guard let tuple = conversationAndOutOfSyncTuples.first(
+                where: { $0.conversation.mlsGroupID?.bytes == groupID }
+            ) else {
+                return 1
+            }
```

Comment:
I think it would be easier to read if the conversations were in a dictionary mapping the group id to the conversation.

Reply:
I agree but I think it would make the setup too complex

Reply:
```swift
        // ...

        let conversationAndOutOfSyncTuples = [
            createConversation(outOfSync: true),
            createConversation(outOfSync: true),
            createConversation(outOfSync: false)
        ]

        let conversationAndOutOfSyncTuplesByGroupID = try conversationAndOutOfSyncTuples.reduce(
            into: [MLSGroupID: ConversationAndOutOfSyncTuple]()
        ) { result, tuple in
            let groupID = try XCTUnwrap(tuple.conversation.mlsGroupID)
            result[groupID] = tuple
        }

        // mock conversation epoch
        mockCoreCrypto.mockConversationEpoch = { groupID in
            guard let tuple = conversationAndOutOfSyncTuplesByGroupID[MLSGroupID(groupID)] else {
                return 1
            }

            let epoch = tuple.conversation.epoch
            return tuple.isOutOfSync ? epoch + 1 : epoch
        }

        // ...
```

Reply:
yeah you're right